crew

```diff
@@ -209,30 +239,26 @@ end
 def resolveDependencies
   @dependencies = []
 
-  def pushDeps
-    if @pkg.dependencies
-      @dependencies.unshift @pkg.dependencies
-      
-      @pkg.dependencies.each do |dep|
-        search dep, true
-        pushDeps
+  if @pkg.dependencies
+    @pkg.dependencies.each do |dep|
+      if not search(dep, true)
+        if not @dependencies.include? dep
+          @dependencies.push(dep)
+        end
       end
     end
   end
-  
-  pushDeps
```

Comment:
Does this pushDeps refactoring works correctly?  What does it happens if A depends_on B, B depends_on C, and C depends_on D?  How did you test your modifications?

It looks like way too much modifications at once for me.